docs/source/concepts/InstrumentAccessLayers.rst

```diff
@@ -31,216 +33,170 @@ ____________
-
-  vector<detid_t> detIds = detectorInfo.detectorIDs();
-
-  for (size_t i = 0; i < detectorInfo.size(); ++i) {
-    if (detectorInfo.isMasked(i)) {
-      maskedDetIds.push_back(detIds[i]);
+  
+  std::set<size_t> componentsHoldingOnlyDetectors;
+  for (size_t i = 0; i < componentInfo.size(); ++i) {
+    if(componentInfo.componentsInSubtree(i) == componentInfo.detectorsInSubtree(i)) {
+     componentsHoldingOnlyDetectors.insert(i); 
     }
   }
 
-Access to Detectors
-___________________
-
-As for the ``SpectrumInfo`` object ``DetectorInfo`` can return a parameterised detector for the workspace.
-
-Mutable DetectorInfo
-____________________
+Mutable ComponentInfo
+_____________________
 
-The method ``ExperimentInfo::mutableDetectorInfo()`` returns a non-const ``DetectorInfo`` object. This allows the methods below to be used.
+The method ``ExperimentInfo::mutableComponentInfo()`` returns a non-const ``ComponentInfo`` object. This allows the methods below to be used.
 
-* ``setMasked(const size_t index, bool masked);``
-* ``clearMaskFlags();``
 * ``setPosition(const size_t index, const Kernel::V3D &position);``
 * ``setRotation(const size_t index, const Kernel::Quat &rotation);``
-* ``setPosition(const Geometry::IComponent &comp, const Kernel::V3D &pos);``
-* ``setRotation(const Geometry::IComponent &comp, const Kernel::Quat &rot);``
-
-For a complete example of setting the position of a detector see the changes to the algorithm `ApplyCalibration.cpp <https://github.com/mantidproject/mantid/commit/0c75f46e85c2dc39c2b76ea811f161fdfdef938e#diff-a4ccabae0099bfc22b3b87154cd6ee9e>`_.
+* ``setScaleFactor(const size_t index, const Kernel::V3D &scaleFactor);``
 
 Useful Tips
 ___________
 
-See tips for ``SpectrumInfo`` - the same advice applies to using ``DetectorInfo``.
-
-Complete Examples
-_________________
-
-* `FindDetectorsInShape.cpp <https://github.com/mantidproject/mantid/commit/177ad14b25db7c40ee10be513512be9ae7dd0da1#diff-7f367da22c1a837b315b4bca5b2b3e24>`_
-
-* `SmoothNeighbours.cpp <https://github.com/mantidproject/mantid/pull/18295/files#diff-26a49ef923e1bdd77677b962528d1e01>`_
-
-* `ClaerMaskFlag.cpp <https://github.com/mantidproject/mantid/pull/18198/files#diff-7f0f739ba6db714eb6ef64b6125e4620>`_
-
-* `ApplyCalibration.cpp <https://github.com/mantidproject/mantid/commit/0c75f46e85c2dc39c2b76ea811f161fdfdef938e#diff-a4ccabae0099bfc22b3b87154cd6ee9e>`_ - for mutable ``DetectorInfo``
-
-Rollout status
---------------
-
-See ticket `17743 <https://github.com/mantidproject/mantid/issues/17743>`_ for an overview of the ``SpectrumInfo`` and ``DetectorInfo`` rollout, including completed and algorithms, and remaining algorithms. Please follow the instructions on the ticket for the rollout.
+* Creation of ``ComponentInfo`` is not cheap enough to perform uncessarily inside loops. For non-const access, ``ws.componentInfo()`` should be called outside of loops that enumerate over all components.
```

Comment:
Actually it is cheap? It is just returned by reference? But I agree with the recommendation.

Also, you mixed up const and non-const, or what you are saying is otherwise unclear?